code/weapon/weapons.cpp

```diff
@@ -6263,6 +6259,13 @@ void weapon_hit( object * weapon_obj, object * other_obj, vec3d * hitpos, int qu
 		emp_apply(&weapon_obj->pos, wip->shockwave.inner_rad, wip->shockwave.outer_rad, wip->emp_intensity, wip->emp_time, (wip->wi_flags[Weapon::Info_Flags::Use_emp_time_for_capship_turrets]) != 0);
 	}	
 
+	// check if this is an electronics weapon
+
+	if (wip->wi_flags[Weapon::Info_Flags::Electronics] && (other_obj != NULL)) {
```

Comment:
This is kind of a minor thing, but the conditional was probably written the other way around because a null pointer check is faster than a flag check, and you generally want to check the fastest conditions first.